src/meepgeom.cpp

```diff
@@ -1057,11 +1113,18 @@ void geom_epsilon::eff_chi1inv_matrix(meep::component c, symm_matrix *chi1inv_ma
                                       bool &fallback) {
   const geometric_object *o;
   material_type mat, mat_behind;
+  vector3 p_mat, p_mat_behind;
   symm_matrix meps;
   vector3 p, shiftby, normal;
+  double fill;
   fallback = false;
+  bool mg_averaging = false;
+  bool center_is_void = true;
 
-  if (maxeval == 0) {
+  if (maxeval == 0 ||
+      (!get_front_object(v, geometry_tree, p, &o, shiftby, mat, mat_behind, p_mat, p_mat_behind)
```

Comment:
I would change `get_front_object` to return the number of objects found (treating the default material as an "object"). 

* If it's `> 2`, then we go to the `noavg:` branch. 
*  If it `== 2`, then even if one of the objects is a material grid or a user material then we treat it as a uniform material (by evaluating at `p_mat` or `p_mat_behind`, respectively) and do the geometric-object averaging (if the resulting `mat` and `mat_behind` are still unequal).   
* If it `== 1`, then if it's a material grid we do the first-order expansion to find the fill fraction etcetera.